src/infra/src/table/distinct_values.rs

```diff
@@ -0,0 +1,218 @@
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+use sea_orm::{
+    entity::prelude::*, ColumnTrait, ConnectionTrait, EntityTrait, FromQueryResult, QueryFilter,
+    Schema, Set,
+};
+
+use super::get_lock;
+use crate::{
+    db::{connect_to_orm, ORM_CLIENT},
+    errors::{self, DbError, Error},
+};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, EnumIter, DeriveActiveEnum)]
+#[sea_orm(rs_type = "String", db_type = "String(StringLen::N(15))")]
+pub enum OriginType {
+    #[sea_orm(string_value = "stream")]
+    Stream,
+    #[sea_orm(string_value = "dashboard")]
+    Dashboard,
+    #[sea_orm(string_value = "report")]
+    Report,
+}
+
+/// Define the distinct fields table
+/// Primary key for this is composite of all fields, i.e.
+/// There will always be only one entry for specific origin-stream-field combination
+#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel)]
+#[sea_orm(table_name = "distinct_value_fields")]
+pub struct Model {
+    #[sea_orm(primary_key, column_type = "Text")]
+    pub origin: OriginType,
+    #[sea_orm(primary_key, column_type = "Text")]
+    pub origin_id: String,
+    #[sea_orm(primary_key, column_type = "Text")]
+    pub org_name: String,
+    #[sea_orm(primary_key, column_type = "Text")]
+    pub stream_name: String,
+    #[sea_orm(primary_key, column_type = "Text")]
+    pub stream_type: String,
+    #[sea_orm(primary_key, column_type = "Text")]
+    pub field_name: String,
+}
+
+#[derive(Copy, Clone, Debug, EnumIter)]
+pub enum Relation {}
+
+impl RelationTrait for Relation {
+    fn def(&self) -> RelationDef {
+        panic!("No relations defined")
+    }
+}
+
+impl ActiveModelBehavior for ActiveModel {}
+
+#[derive(FromQueryResult, Debug)]
+pub struct DistinctFieldRecord {
+    pub origin: OriginType,
+    pub origin_id: String,
+    pub org_name: String,
+    pub stream_name: String,
+    pub stream_type: String,
+    pub field_name: String,
+}
+
+impl DistinctFieldRecord {
+    pub fn new(
+        origin: OriginType,
+        origin_id: &str,
+        org: &str,
+        stream: &str,
+        stream_type: String,
+        field: &str,
+    ) -> Self {
+        Self {
+            origin,
+            origin_id: origin_id.to_owned(),
+            org_name: org.to_owned(),
+            stream_name: stream.to_owned(),
+            stream_type,
+            field_name: field.to_owned(),
+        }
+    }
+}
+
+pub async fn init() -> Result<(), errors::Error> {
+    create_table().await?;
+    Ok(())
+}
+
+pub async fn create_table() -> Result<(), errors::Error> {
```

Comment:
If we want to make use of Sea ORM's [migration functionality](https://github.com/openobserve/openobserve/pull/5146) we might want to consider creating a dedicated and timestamped migration file for this in the `/infra/src/table/migrations` folder so that we can ensure that this migration always gets executed in the correct order relative to other migrations.

Reply:
Hey so originally I had tried using sea orm migrations, but I ran into some issues while accessing the dashboard types ( v1::Dashboard to v5::Dashboard ) in the crate where migrations are defined. So I reverted back to this approach. This is definitely going to run into the issue of order, ie sea orm migration might run before or after this ; which is partly why I'm waiting on you PR so I can leverage that to access to the new dashboard table and double check all dashboards are migrated. That code will not be pretty, but for a one-off migration I feel it will be ok to have double checks.

Reply:
> but I ran into some issues while accessing the dashboard types ( v1::Dashboard to v5::Dashboard ) in the crate where migrations are defined

I did end up moving those struct definitions into the `config` crate, so I think they should be accessible from the `infra` crate now without introducing a circular dependency.

I'm not sure if this is what you're suggesting but I think it might be worth pointing out that (personally) as a general principle I think it's safest if each migration script is a self-contained script without referencing external data types, because changes to those data types can result in unintentional changes to the migration script behavior and compromise the consistent reproducibility of migrations. So although I could certainly see it being useful to access those data types from "DB access function", I'd personally be wary of referencing those types from within each migration script itself.